bluebrain/repo-bluebrain/packages/neurodamus-models/package.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class NeurodamusModels(CMakePackage):
+    """Neuroscientific models to be used with Neurodamus"""
+
+    homepage = "https://github.com/BlueBrain/neurodamus-models"
+    git = "ssh://git@github.com/BlueBrain/neurodamus-models.git"
+
+    version("develop", branch="cfaire")
+    version("6.6.6", commit="d66d0de8741543497bb127998deba7a5e3f7de42")
+    version("6.6.7", commit="8e21f8e16e037d0f1042d4585b1417f083bd1cea")
+    version("6.6.8", commit="506f1cb7dbf59d9dfca835245597e59e144ed24a")
+    version("6.6.9", commit="2cc59da34d22633df5baff3a2ec282e93de54b97")
+    version("6.7.0", commit="0eee2cc6a4d8f84f3f7980fc6d9c88f5a5f258df")
+    version("6.7.1", commit="3ae2eb94fc8f415d32b89b4003d12b24f4d03384")
+    version("6.7.2", commit="4d89b24a46a36e2b9f1f2d2ee108d7f1c478e64f")
+    version("6.7.3", commit="2dc61b533c1b036c3b52ed0aa14393932d269842")
+    version("6.7.4", commit="b848572f214f799ccb0ccb8615ceb8b925f6dca9")
+
+    variant("caliper", default=False, description="Enable Caliper instrumentation")
+    variant("coreneuron", default=False, description="Enable CoreNEURON support")
+
+    variant(
+        "model",
+        default="neocortex",
+        values=("hippocampus", "neocortex", "thalamus"),
+        multi=False,
+        description="Which mechanisms to enable",
+    )
+    variant("ngv", default=False, when="model=neocortex", description="Enable NGV mechanisms")
+    variant(
+        "metabolism", default=False, when="model=neocortex", description="Enable NGV mechanisms"
+    )
+    variant(
+        "plasticity", default=False, when="model=neocortex", description="Enable NGV mechanisms"
+    )
+    variant("v5", default=True, when="model=neocortex", description="Enable NGV mechanisms")
+
+    depends_on("neuron", type=("build", "link", "run"))
+    depends_on("neuron+coreneuron", when="+coreneuron", type=("build", "link", "run"))
+    depends_on("neuron+caliper", when="+caliper", type=("build", "link", "run"))
+
+    depends_on("py-neurodamus", type=("build", "run"))
+    depends_on("libsonata-report")
+
+    def cmake_args(self):
+        args = [
+            self.define("ND_CORE_DIR", self.spec["py-neurodamus"].prefix.lib),
+            self.define_from_variant("ND_MECHANISMS", "model"),
+            self.define_from_variant("ND_ENABLE_CORENEURON", "coreneuron"),
+            self.define_from_variant("ND_NCX_METABOLISM", "metabolism"),
+            self.define_from_variant("ND_NCX_NGV", "ngv"),
+            self.define_from_variant("ND_NCX_PLASTICITY", "plasticity"),
+            self.define_from_variant("ND_NCX_V5", "v5"),
```

Comment:
@matz-e: Relatively minor but I am wondering about your opinion here: while reviewing BlueBrain/neurodamus-models/pull/3, I thought having a bit more descriptive name is better than abbreviation. E.g. In this case, `NEURODAMUS_ FOO` is better than `ND_FOO`?

It's a bit verbose but clearer? Once we start using these cmake options in other places like this, later they are cumbersome to change and hence this comment.

Reply:
OK, will do that.